tracer/src/Datadog.Trace/Iast/IastModule.cs

```diff
@@ -810,6 +831,86 @@ internal static void OnEmailHtmlInjection(object? message)
         }
 
         // We use the same secure marks as XSS
-        GetScope(messageDuck.Body, IntegrationId.EmailHtmlInjection, VulnerabilityTypeName.EmailHtmlInjection, OperationNameEmailHtmlInjection, taintValidator: Always, exclusionSecureMarks: SecureMarks.Xss);
+        GetScope(messageDuck.Body, IntegrationId.EmailHtmlInjection, VulnerabilityTypeName.EmailHtmlInjection, OperationNameEmailHtmlInjection, taintValidator: NoDbSource, exclusionSecureMarks: SecureMarks.Xss);
+    }
+
+    internal static void RegisterDbRecord(object instance)
+    {
+        DbRecords.RegisterDbRecord(instance);
+    }
+
+    internal static void UnregisterDbRecord(object instance)
+    {
+        DbRecords.UnregisterDbRecord(instance);
+    }
+
+    internal static bool AddDbValue(object instance, string? column, string value)
+    {
+        return DbRecords.AddDbValue(instance, column, value);
+    }
+
+    internal class DbRecordManager
+    {
+        private ConditionalWeakTable<object, DbRecordData> dataBaseRows = new ConditionalWeakTable<object, DbRecordData>();
+        private IastSettings iastSettings;
+
+        public DbRecordManager(IastSettings settings)
+        {
+            iastSettings = settings;
+        }
+
+        public void RegisterDbRecord(object instance)
+        {
+            if (!iastSettings.Enabled || iastSettings.DataBaseRowsToTaint <= 0)
+            {
+                return;
+            }
+
+            var recordData = dataBaseRows.GetOrCreateValue(instance);
+            recordData.Count++;
+        }
+
+        public void UnregisterDbRecord(object instance)
+        {
+            if (!iastSettings.Enabled || iastSettings.DataBaseRowsToTaint <= 0)
+            {
+                return;
+            }
+
+            dataBaseRows.Remove(instance);
+        }
+
+        public bool AddDbValue(object instance, string? column, string value)
+        {
+            if (!iastSettings.Enabled || iastSettings.DataBaseRowsToTaint <= 0)
+            {
+                return false;
+            }
+
+            if (iastSettings.DataBaseRowsToTaint > 0)
+            {
+                if (!dataBaseRows.TryGetValue(instance, out var recordData))
+                {
+                    return false;
+                }
+
+                if (recordData.Count > iastSettings.DataBaseRowsToTaint)
+                {
+                    return false;
+                }
+            }
+
+            var tracer = Tracer.Instance;
+            var scope = tracer.ActiveScope as Scope;
+            var currentSpan = scope?.Span;
+            var traceContext = currentSpan?.Context?.TraceContext;
+            traceContext?.IastRequestContext?.AddDbValue(column, value);
+            return true;
```

Comment:
because of the null coercion we could have failed to add the data, but we'd still return true.

Reply:
I'm not sure if the return value is used?